src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -582,13 +583,49 @@ object Lorre extends App with TezosErrors with LazyLogging with LorreAppConfig w
       def processAccountsPage(
           taggedAccounts: List[BlockTagged[AccountsIndex]],
           taggedDelegateKeys: List[BlockTagged[DelegateKeys]]
-      ): Future[(Option[Int], Option[Int], List[BlockTagged[DelegateKeys]])] =
-        db.run(TezosDb.writeAccountsAndCheckpointDelegates(taggedAccounts, taggedDelegateKeys))
-          .map {
+      ): Future[(Option[Int], Option[Int], List[BlockTagged[DelegateKeys]])] = {
+        for {
+          activatedOperations <- fetchActivatedOperationsByLevel(taggedAccounts.map(_.blockLevel).distinct)
+          activatedAccounts <- db.run(TezosDb.findActivatedAccounts)
+          updatedTaggedAccounts = updateTaggedAccountsWithIsActivated(
+            taggedAccounts,
+            activatedOperations,
+            activatedAccounts.toList
+          )
+          res <- db.run(TezosDb.writeAccountsAndCheckpointDelegates(updatedTaggedAccounts, taggedDelegateKeys)).map {
             case (accountWrites, accountHistoryWrites, delegateCheckpoints) =>
               (accountWrites, delegateCheckpoints, taggedDelegateKeys)
           }
-          .andThen(logWriteFailure)
+        } yield res
+      }.andThen(logWriteFailure)
+
+      def updateTaggedAccountsWithIsActivated(
+          taggedAccounts: List[BlockTagged[AccountsIndex]],
+          activatedOperations: Map[Int, Seq[Tables.OperationsRow]],
+          activatedAccountIds: List[String]
+      ): List[BlockTagged[Map[AccountId, Account]]] =
+        taggedAccounts.map { taggedAccount =>
+          val activatedAccountsHashes = activatedOperations.get(taggedAccount.blockLevel).toSeq.flatten.map(_.pkh)
+          taggedAccount.copy(
+            content = taggedAccount.content.mapValues { account =>
+              val hash = account.manager.map(_.value)
+              if (activatedAccountsHashes.contains(hash) || activatedAccountIds.contains(hash.getOrElse(""))) {
+                account.copy(isActivated = Some(true))
+              } else account
+            }
+          )
+        }
+
+      def fetchActivatedOperationsByLevel(levels: List[Int]): Future[Map[Int, Seq[Tables.OperationsRow]]] = {
+        import slick.jdbc.PostgresProfile.api._
+        db.run {
+          DBIO.sequence {
+            levels.map { level =>
+              TezosDb.fetchRecentOperationsByKind(Set("activate_account"), level).map(x => level -> x)
```

Comment:
Would it be an improvement if this function already returned the pkhs instead of a whole row?

Reply:
sure, changed